### DISCUSSION TITLE
Logging User Question+ChatFlowID+Answer

### DISCUSSION THREAD

**Discussion by marius-smg** [FlowiseAI/Flowise#723] [Q&A]
Hey, for analysing the usage of my chatbots i want to track the questions that are being asked, the IDs and the answeres that are generated. I know those parameters are already logged but not in a way that they are easy readable. It should be possible to make an extra log entry containing those infos. After that i can easily search in the Log stream just for those specific log entries. I just cant find the right place where to implement this myself. Can anyone point me to the right file i'm looking for? If You have any further tipps, help is much appreciated.

**Comment by chungyau97** [comment]
Hey @marius-smg,

Currently we are still developing the [chat history](https://github.com/FlowiseAI/Flowise/pull/715) for internal and external chat. 
Then the next stage will allow owner to export chat history for internal and external chat.
